### PR TITLE
Repository issues view

### PR DESCRIPTION
**Pull request by KaiwalyaPataskar**
Here we've fetched repositories from CodeClimate and displayed in the list form along with their Maintainability & Test Coverage badges, last activity timestamp etc.
On drill down, we're showing all issues fetched from CodeClimate along with their Category, Description, Severity, File path & Line number.

Screenshot from 2020-05-14 19-59-52

image

### CODE REVIEW

File: app/views/projects/_repository_form.html.haml [hunk 1]
```diff
@@ -19,9 +19,16 @@
             = t.input :maintainability_badge, placeholder: "Code Climate -> Repo Settings"
           .span4
             = t.input :test_coverage_badge, placeholder: "Code Climate -> Repo Settings"
-          .span2{style: "float: right;"}
+        .row
+          .span2
+            = t.label "visibility"
```
Comment (anujaware) [inline app/views/projects/_repository_form.html.haml:24]: @KaiwalyaPataskar Just confirm that this label is self explanatory while using the feature by end user.

File: app/views/projects/_team_info_form.html.haml [hunk 2]
```diff
@@ -13,25 +13,17 @@
       - if project_user_form.object.user.blank? || (project_user_form.object.user && project_user_form.object.user.is_approved? && (project_user_form.object.active || project_user_form.object.end_date.nil?))
         .row.span12
           .span2.width12
-            - if project_user_form.object.user.blank?
-              .new-switch{"data-on" => "success", "data-off" => "danger", "data-on-label" => "Yes", "data-off-label" => "No"}
-                = project_user_form.check_box :active, {class: "active-check-box"}, 'true', 'false'
```
Comment (anujaware) [inline app/views/projects/_team_info_form.html.haml:18]: @KaiwalyaPataskar Is it relevant change?

File: app/views/repositories/repository_issues.html.haml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        %th Line Number
+    %tbody
+      %script#issues_template{:type => "text/html"}
+        %tr
+          {{#record}}
+          %td
+            {{index}}
+          %td
+            {{attributes.description}}
+          %td{style:"color:{{color}}; font-weight:bold; text-transform:capitalize;"}
```
Comment (anujaware) [inline app/views/repositories/repository_issues.html.haml:26]: @KaiwalyaPataskar Remove inline css.

File: app/assets/javascripts/stream_table.min.js [hunk 4]
```diff
@@ -1,13 +0,0 @@
...
...:s[0]};t.sorting_opts[a.field]={dir:a.dir,type:a.type,field:a.field},i.on("click",a,function(e){var i=$(this);i.addClass(e.data.dir),t.current_sorting={dir:e.data.dir,type:e.data.type,field:e.data.field},t.sort(e.data),t.render(t.current_page),e.data.dir="asc"==e.data.dir?"desc":"asc",$(this).removeClass(e.data.dir),t.opts.callbacks.after_sort&&t.execCallbacks("after_sort")}),0==e&&t.opts.auto_sorting&&i.trigger("click")})},_F.sort=function(t){return t.order="asc"==t.dir?1:-1,this.getIndex().sort(this._sortingFunc(this.data,t))},_F._sortingFunc=function(t,e){var i=e.field,s=e.order,a=e.type;return"number"==a?function(e,a){return(t[e][i]-t[a][i])*s}:function(e,a){var n=t[e][i].toLowerCase(),r=t[a][i].toLowerCase();return r>n?-1*s:n>r?1*s:0}},_F.clear=function(){this.opts.search_box&&$(this.opts.search_box).html(""),$(this.main_container).html("")},StreamTable.extend=function(t,e){_StreamTable.prototype[t]=function(){return e.apply(this,arguments)}}}(this,window.jQuery),Array.prototype.indexOf||(Array.prototype.indexOf=function(t,e){for(var i=e||0,s=this.length;s>i;i++)if(this[i]===t)return i;return-1}),Object.keys||(Object.keys=function(t){var e,i=[];for(e in t)i.push(e);return i});
```
[the remark this answers is not in the document]
Comment (KaiwalyaPataskar) [reply app/assets/javascripts/stream_table.min.js:13]: Yes, we should do that.

File: app/views/projects/_team_info_form.html.haml (hunk 2 above)
Comment (KaiwalyaPataskar) [reply app/views/projects/_team_info_form.html.haml:18]: This is a just a bug fix.
Later @abhi-sapate opened a PR for the same but it's still Open!
Link: https://github.com/joshsoftware/intranet/pull/382

File: app/assets/stylesheets/repositories.css.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
+.severity {
```
Comment (anujaware) [inline app/assets/stylesheets/repositories.css.scss:1]: @KaiwalyaPataskar let's not add new file for generic and few items. Add it to common file. Also confirm if it is already present.

**Comment by anujaware** [comment]
@KaiwalyaPataskar Is it done? Please assign the PR to me once you are done.
@swapnilchincholkar Please review the PR.

**Comment by anujaware** [comment]
@swapnilchincholkar You can review this PR. Please check and let me know if we can plan to release it in this week.

**Comment by anujaware** [comment]
@swapnilchincholkar We can merge this PR and release it on staging for testing.
cc: @KaiwalyaPataskar

**Pull request merged**